.circleci/config.yml

```diff
@@ -229,15 +214,18 @@ references:
     paths:
       - packages
       - packages-cache-restored
-  desktop-restore-webpack-css-loader-cache: &desktop-restore-webpack-css-loader-cache
-    name: Restore webpack cache-loader cache
+  desktop-restore-calypso-build-cache: &desktop-restore-calypso-build-cache
+    name: Restore Calypso build cache
     keys:
-      - v6-desktop-webpack-css-loader-{{ .Environment.CIRCLE_JOB }}-{{ arch }}-{{ checksum "webpack-css-loader-hash" }}
-  desktop-save-webpack-css-loader-cache: &desktop-save-webpack-css-loader-cache
-    name: Save webpack cache-loader cache
-    key: v6-desktop-webpack-css-loader-{{ .Environment.CIRCLE_JOB }}-{{ arch }}-{{ checksum "webpack-css-loader-hash" }}
+      - v7-desktop-calypso-build-{{ .Environment.CIRCLE_JOB }}-{{ arch }}-{{ .Branch }}-{{ .Revision }}
+      - v7-desktop-calypso-build-{{ .Environment.CIRCLE_JOB }}-{{ arch }}-{{ .Branch }}
+      - v7-desktop-calypso-build-{{ .Environment.CIRCLE_JOB }}-{{ arch }}-trunk
+      - v7-desktop-calypso-build-{{ .Environment.CIRCLE_JOB }}-{{ arch }}
+  desktop-save-calypso-build-cache: &desktop-save-calypso-build-cache
+    name: Save Calypso build cache
+    key: v7-desktop-calypso-build-{{ .Environment.CIRCLE_JOB }}-{{ arch }}-{{ .Branch }}-{{ .Revision }}
```

Comment:
Optional, but I'd consider adding the environment `GLOBAL_CACHE_PREFIX` here as well. For context, some things in the cache seem to have unbounded growth. The environment variable can be managed by CircleCI UI so makes it trivial to flush caches without needing to commit anything.

It makes sense to flush caches periodically because at a certain point it's worth it to flush them once to reduce the size of what we store and retrieve frequently throughout the process.

```suggestion
      - v{{ .Environment.GLOBAL_CACHE_PREFIX }}-v0-desktop-calypso-build-{{ .Environment.CIRCLE_JOB }}-{{ arch }}-{{ .Branch }}-{{ .Revision }}
      - v{{ .Environment.GLOBAL_CACHE_PREFIX }}-v0-desktop-calypso-build-{{ .Environment.CIRCLE_JOB }}-{{ arch }}-{{ .Branch }}
      - v{{ .Environment.GLOBAL_CACHE_PREFIX }}-v0-desktop-calypso-build-{{ .Environment.CIRCLE_JOB }}-{{ arch }}-trunk
      - v{{ .Environment.GLOBAL_CACHE_PREFIX }}-v0-desktop-calypso-build-{{ .Environment.CIRCLE_JOB }}-{{ arch }}
  desktop-save-calypso-build-cache: &desktop-save-calypso-build-cache
    name: Save Calypso build cache
    key: v{{ .Environment.GLOBAL_CACHE_PREFIX }}-v0-desktop-calypso-build-{{ .Environment.CIRCLE_JOB }}-{{ arch }}-{{ .Branch }}-{{ .Revision }}
```

Even better would be to automate that to clear caches every month with something like:

```sh
# Run as part of prepare step
date +%Y-%m > ci-cache-prefix.txt
```

Use it as a prefix:
```yml
      - {{ checksum "ci-cache-prefix.txt" }}-desktop-packages-{{ .Environment.CIRCLE_JOB }}-{{ arch }}-{{ checksum "packages-hash" }}
```

Reply:
@sirreal Your suggestion makes sense - thanks for pointing this out. However I'm a little confused by the suggested change wrt to the sequence of restoration keys:

```
      - v{{ .Environment.GLOBAL_CACHE_PREFIX }}-v0-desktop-calypso-build-{{ .Environment.CIRCLE_JOB }}-{{ arch }}-{{ .Branch }}
      - v{{ .Environment.GLOBAL_CACHE_PREFIX }}-v0-desktop-calypso-build-{{ .Environment.CIRCLE_JOB }}-{{ arch }}-trunk
```

`trunk` is used here while not being part of the longer prefix used prior (either for saving or restoring). My understanding is this will effectively be a no-op as it eventually matches against `- v{{ .Environment.GLOBAL_CACHE_PREFIX }}-v0-desktop-calypso-build-{{ .Environment.CIRCLE_JOB }}-{{ arch }}`. Is the usage of `trunk` here intentional?

Reply:
Edit: I suppose this might be a question for @scinos, as I see the usage of trunk was introduced prior to the suggested change.^

Reply:
It is intentional. `trunk` is replacing `master`. It is (or will be) the default branch and a good assumption is that it's the next best thing if we can't match the branch we're on.

> My understanding is this will effectively be a no-op as it eventually matches against

I don't completely understand your question. These caches find the first matching candidate and use that. So the `…-trunk` candidate will match a recent cache like `…-trunk-abc123`. It's likely that subsequent candidates will not be reached.